src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -261,7 +261,7 @@ The term is composed of three parts
   # Compute charge ratio of u_ll
   #
   charge_ratio_ll = zeros(MVector{ncomponents(equations), eltype(u_ll)})
-  total_electron_charge = zero(u_ll[1])
+  total_electron_charge = 0
```

Comment:
```suggestion
  total_electron_charge = zero(u_ll[1])
```
Here you should keep the original formulation to retain type stable code (cc @ranocha)